core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -396,6 +393,26 @@ public static boolean isCorsPreflightRequest(com.linecorp.armeria.common.HttpReq
                request.headers().contains(HttpHeaderNames.ACCESS_CONTROL_REQUEST_METHOD);
     }
 
+    /**
+     * Returns whether the specified header name is disallowed for additional trailers.
+     */
+    public static boolean isDisallowedAdditionalTrailer(AsciiString name) {
+        // Pseudo headers are not allowed for additional trailers.
+        return !name.isEmpty() && name.charAt(0) == ':';
+    }
+
+    /**
+     * Returns whether the specified header name is disallowed for response headers.
+     */
+    public static boolean isDisallowedResponseHeader(AsciiString name) {
+        // Request Pseudo-Headers are not allowed for response headers.
+        // https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.3
+        if (HttpHeaderNames.STATUS.equals(name)) {
+            return false;
+        }
+        return isDisallowedAdditionalTrailer(name);
```

Comment:
We could check this only when the first character is `:` for efficiency.
```suggestion
        if (isDiallowedAdditionalTrailer(name)) {
            return !HttpHeaderNames.STATUS.equals(name);
        } else {
            return true;
        }
```